cmake/modules/FindFlatBuffers.cmake

```diff
@@ -0,0 +1,72 @@
+# FindFlatBuffers
+# --------
+# Find the FlatBuffers schema compiler and headers
+#
+# This will define the following variables:
+#
+# FLATBUFFERS_FOUND - system has FlatBuffers compiler and headers
+# FLATBUFFERS_FLATC_EXECUTABLE - the flatc compiler executable
+# FLATBUFFERS_INCLUDE_DIRS - the FlatFuffers include directory
+# FLATBUFFERS_MESSAGES_INCLUDE_DIR - the directory for generated headers
+
+if(ENABLE_INTERNAL_FLATBUFFERS)
+  include(ExternalProject)
+  file(STRINGS ${CMAKE_SOURCE_DIR}/tools/depends/native/flatbuffers-native/Makefile VER REGEX "^[ ]*VERSION[ ]*=.+$")
+  string(REGEX REPLACE "^[ ]*VERSION[ ]*=[ ]*" "" FLATBUFFERS_VER "${VER}")
+
+  # Allow user to override the download URL with a local tarball
+  # Needed for offline build envs
+  if(FLATBUFFERS_URL)
+    get_filename_component(FLATBUFFERS_URL "${FLATBUFFERS_URL}" ABSOLUTE)
+  else()
+    set(FLATBUFFERS_URL http://mirrors.kodi.tv/build-deps/sources/flatbuffers-${FLATBUFFERS_VER}.tar.gz)
+  endif()
+  if(VERBOSE)
+    message(STATUS "FLATBUFFERS_URL: ${FLATBUFFERS_URL}")
+  endif()
+
+  set(FLATBUFFERS_FLATC_EXECUTABLE ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/bin/flatc CACHE INTERNAL "FlatBuffer compiler")
+  set(FLATBUFFERS_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include CACHE INTERNAL "FlatBuffer include dir")
+  set(FLATBUFFERS_MESSAGES_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/cores/RetroPlayer/messages CACHE INTERNAL "Generated FlatBuffer headers")
+
+  externalproject_add(flatbuffers
+                      URL ${FLATBUFFERS_URL}
+                      DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/download
+                      PREFIX ${CORE_BUILD_DIR}/flatbuffers
+                      CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}
+                                 -DCMAKE_BUILD_TYPE=Release
```

Comment:
@wsnipex @Rechi Debug build fails with internal flatbuffers. `flatc.exe` is not installed for debug builds because debug symbols don't make sense for the compiler, so we need to force it to build in Release mode always.

CMake is ignoring this flag when I select "Debug" in Visual Studio. Is there a way to override cmake's override?

Reply:
`flatc.exe`, so you try to use the internal build for Windows? Why?

Reply:
I assumed we wanted correctness for both modes when building internal. Is this not the case?

Reply:
so it's ok that we can't build debug with internal flatbuffers?

Reply:
Wasn't I clear enough, why would you want to use internal flatbuffers on windows if you have to use all other precompiled dependencies anyway?

Reply:
Thanks for explaining